scipy/spatial/meson.build

```diff
@@ -0,0 +1,112 @@
+_spatial_pxd = custom_target('_spatial_pxd',
+  output : ['qhull.pxd', 'setlist.pxd'],
+  input : ['qhull.pxd', 'setlist.pxd'],
+  command : ['cp', '@INPUT@', '-t', '@OUTDIR@'],
+)
+
+qhull_src = ['qhull_src/src/poly_r.c',
+'qhull_src/src/stat_r.c',
+'qhull_src/src/merge_r.c',
+'qhull_src/src/userprintf_r.c',
+'qhull_src/src/userprintf_rbox_r.c',
+'qhull_src/src/usermem_r.c',
+'qhull_src/src/geom_r.c',
+'qhull_src/src/libqhull_r.c',
+'qhull_src/src/user_r.c',
+'qhull_src/src/random_r.c',
+'qhull_src/src/geom2_r.c',
+'qhull_src/src/mem_r.c',
+'qhull_src/src/global_r.c',
+'qhull_src/src/poly2_r.c',
+'qhull_src/src/io_r.c',
+'qhull_src/src/rboxlib_r.c',
+'qhull_src/src/qset_r.c']
+
+qhull = py3.extension_module('qhull',
+  ['qhull.pyx', 'qhull_misc.h', 'qhull_misc.c'] + qhull_src,
+  c_args: '-Wno-cpp',
+  include_directories: [incdir_numpy, '../_lib', '../_build_utils/src', 'qhull_src/src'],
+  dependencies: [py3_dep, lapack],
```

Comment:
Absence of `lapack` isn't reflected during build, but an error is raised (see Traceback below) when we run the tests of `spatial`. So I added it here. 

```bash
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/home/czgdp1807ssd/scipy_project/meson_build/install/lib/python3.9/site-packages/scipy/spatial/__init__.py", line 98, in <module>
    from .qhull import *
ImportError: /home/czgdp1807ssd/scipy_project/meson_build/install/lib/python3.9/site-packages/scipy/spatial/qhull.cpython-39-x86_64-linux-gnu.so: undefined symbol: dgecon_
```

Reply:
Yes, that happens a lot with LAPACK - problems can show up post-build. Note that `setup.py` has this too, but it's a bit hard to read:
```python
cfg = combine_dict(lapack_opt, include_dirs=inc_dirs)
config.add_extension('qhull',
    ...
    **cfg)
```
This same pattern is used in a bunch of places.

Reply:
Yeah. I have noticed this while porting the code from setuptools to meson. It seems like we can face this problem but it goes away once we include `lapack` as a dependency.